Fortinet/fortigate/ingest/parser.yml

```
@@ -0,0 +1,230 @@
+name: fortigate
+pipeline:
+  - name: parsed_event
+    external:
+      name: cef.parse-cef
+    filter: '{{original.message.startswith("CEF")}}'
+
+  - name: parsed_event
+    external:
+      name: kv.parse-kv
+      properties:
+        value_sep: '='
+        item_sep: ','
+    filter: "{{original.message | re_match('^\\s*\\w+=[^,]+(?:,\\w+=.*)\\s*$')}}"
+
+  - name: parsed_event
+    external:
+      name: kv.parse-kv
+      properties:
+        value_sep: '='
+        item_sep: \s
+    filter: "{{original.message | re_match('^\\s*\\w+=[^\\s]+(?:\\s\\w+=.*)\\s*$')}}"
+
+  - name: field_extraction
+
+stages:
+  field_extraction:
+    actions:
+      - set:
+          action.name: "{{parsed_event.message.name or parsed_event.message.FTNTFGTaction or parsed_event.message.FortinetFortiGateaction or parsed_event.message.act or parsed_event.message.action}}"
+          destination.ip: "{{parsed_event.message.dstip or parsed_event.message.dst}}"
+          destination.address: "{{parsed_event.message.dstip or parsed_event.message.dst}}"
+          source.ip: "{{parsed_event.message.remip or parsed_event.message.srcip or parsed_event.message.src}}"
+          source.address: "{{parsed_event.message.remip or parsed_event.message.srcip or parsed_event.message.src}}"
+          destination.mac: "{{parsed_event.message.dstmac}}"
+          destination.port: "{{parsed_event.message.dstport or parsed_event.message.dpt}}"
+          destination.domain: "{{parsed_event.message.hostname}}"
+          log.level: "{{parsed_event.message.level or parsed_event.message.FTNTFGTlevel or parsed_event.message.FortinetFortiGatelevel}}"
+          log.description: "{{parsed_event.message.logdesc}}"
+          event.code: "{{parsed_event.message.logid or parsed_event.message.FTNTFGTlogid or parsed_event.message.FortinetFortiGatelogid}}"
+          event.type: "{{parsed_event.message.desc}}"
+          event.reason: "{{parsed_event.message.reason}}"
```

```suggestion
          event.action: "{{parsed_event.message.reason}}"
```

This field looks more like [event.action](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-action) to me